### PR TITLE
[SPARK-10626][MLLIB] create java friendly method for random rdd

### PR DESCRIPTION
**Pull request by holdenk**
SPARK-3136 added a large number of functions for creating Java RandomRDDs, but for people that want to use custom RandomDataGenerators we should make a Java friendly method.

### CODE REVIEW

File: mllib/src/test/java/org/apache/spark/mllib/random/JavaRandomRDDsSuite.java [hunk 1]
```diff
@@ -231,4 +232,30 @@ public void testGammaVectorRDD() {
...
+    for (JavaRDD<String> rdd: Arrays.asList(rdd1, rdd2, rdd3)) {
+      Assert.assertEquals(size, rdd.count());
+      Assert.assertEquals(2, rdd.first().length());
+    }
+  }
+}
+
+// This is just a test generator, it always returns a string of 42
+class StringGenerator implements RandomDataGenerator<String>, Serializable {
+  @Override public String nextValue() {
```
Comment (srowen) [inline mllib/src/test/java/org/apache/spark/mllib/random/JavaRandomRDDsSuite.java]: This is all nit picking but I think `@Override` and other annotations go on a separate line at least in the Spark code I see. The last two methods miss a space before the brace. You might sort `java.io` imports earlier.

**Comment by srowen** [comment]
LGTM because it looks like the other Scala RDD utility methods have Java parallels here too.
How about the random vector RDD method too?

**Comment by holdenk** [comment]
@srowen I figured providing a general method would be easier than playing whack-a-mole with all of the other wrapper methods, but I can either add vector to this one or do it in a separate PR (preference?)

**Comment by srowen** [comment]
I don't have a strong opinion on it. It looks like there is this parallel structure for everything except the random and random-vector methods, for some reason. But yeah I can see that this is getting pretty heavy. I leave it to your taste.

**Comment by holdenk** [comment]
So I created https://issues.apache.org/jira/browse/SPARK-10706 to add random vector wrapper for the Java API as well and I'll follow up with that in a bit.

**Comment by srowen** [comment]
I'm looking at https://github.com/apache/spark/pull/8841 now, for comparison, and noted that the Java overloads there call the scala overloads directly, rather than call the primary Java method with default values. The former is how the rest of the class seems to work. Maybe that's better? consolidates defaults like 0 for numPartitions in one place.

Is the implicit ctag needed? may be so, just didn't see it in other java methods.

**Comment by holdenk** [comment]
@srowen : So the implicit tag is needed because the signature for this one is different (all the other ones are calling concrete types and this is a templated type).

For calling the defaults I did it this way since we can share the class tag creation this way, but if you want I can swap it to be like the others.

**Comment by srowen** [comment]
Sounds good, there's a logic to it then, gotcha. I figured as much. Leave it, of course.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**